pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -1719,60 +1756,62 @@ func validatePodDNSConfig(dnsConfig *k8sv1.PodDNSConfig, dnsPolicy *k8sv1.DNSPol
 		}
 	}
 
-	if dnsConfig != nil {
-		// Validate nameservers.
-		if len(dnsConfig.Nameservers) > maxDNSNameservers {
+	if dnsConfig == nil {
+		return causes
+	}
+	// Validate nameservers.
+	if len(dnsConfig.Nameservers) > maxDNSNameservers {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("must not have more than %v nameservers: %s", maxDNSNameservers, dnsConfig.Nameservers),
+			Field:   "nameservers",
+		})
+	}
+	for _, ns := range dnsConfig.Nameservers {
+		if ip := net.ParseIP(ns); ip == nil {
 			causes = append(causes, metav1.StatusCause{
 				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: fmt.Sprintf("must not have more than %v nameservers: %s", maxDNSNameservers, dnsConfig.Nameservers),
+				Message: fmt.Sprintf("must be valid IP address: %s", ns),
 				Field:   "nameservers",
 			})
 		}
-		for _, ns := range dnsConfig.Nameservers {
-			if ip := net.ParseIP(ns); ip == nil {
-				causes = append(causes, metav1.StatusCause{
-					Type:    metav1.CauseTypeFieldValueInvalid,
-					Message: fmt.Sprintf("must be valid IP address: %s", ns),
-					Field:   "nameservers",
-				})
-			}
-		}
-		// Validate searches.
-		if len(dnsConfig.Searches) > maxDNSSearchPaths {
+	}
+	// Validate searches.
```

Comment:
this long long section begs to be extracted, in a follow-up, to a `validateDnsSearches` helper function